mountpoint-s3/src/data_cache/express_data_cache.rs

```diff
@@ -102,7 +120,12 @@ where
         block_idx: BlockIndex,
         block_offset: u64,
         bytes: ChecksummedBytes,
+        object_size: usize,
     ) -> DataCacheResult<()> {
+        if object_size > self.max_object_size {
```

Comment:
We may also want to skip the upload, if a block is already in the shared cache. This could be a more common pattern than with the local cache. Does this optimisation worth an extra `HeadObject` request with each `put_object`?

Reply:
Might also be worth checking if we can do `put_if_none_match` headers and when they validate. Might be able to get away with not too many re-uploads

Reply:
But would `put_if_none_match` actually save anything?

Reply:
Good call, but I don't expect [If-None-Match](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutObject.html#API_PutObject_RequestSyntax) header to result in an optimisation here. I'd expect the check of object existence to happen only after we've uploaded the body of the http request. [Documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/conditional-writes.html) doesn't say anything regarding that.